src/rgw/rgw_file.h

```diff
@@ -802,7 +802,10 @@ namespace rgw {
 			     const uint32_t flags = RGWFileHandle::FLAG_NONE) {
       using std::get;
 
-      LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      // This seems like a Clang compiler error.
+      // It result in irgw::RGWFileHandle::FLAG_NONE being undefined during linking.
+      // LookupFHResult fhr { nullptr, RGWFileHandle::FLAG_NONE };
+      LookupFHResult fhr { nullptr, 0 };
```

Comment:
@tchaikov The Clang 3.8 output without the rewite for FLAG_NONE is:
libtool: link: clang++ -Wall -Wtype-limits -Wignored-qualifiers -Winit-self -Wpointer-arith -Werror=format-security -fno-strict-aliasing -fsigned-char -ftemplate-depth-1024 -Wnon-virtual-dtor -Wno-invalid-offsetof -g -DGTEST_USE_OWN_TR1_TUPLE=1 -I/usr/local/include -Wno-unused-function -Wno-unused-local-typedef -Wno-inconsistent-missing-override -Wno-unused-private-field -Wno-varargs -Wno-gnu-designator -Wno-mismatched-tags -std=gnu++11 -g -o .libs/ceph_rgw_multiparser rgw/rgw_multiparser.o -Wl,--export-dynamic  -L/usr/local/lib ./.libs/librgw.so /usr/srcs/Ceph/work/ceph/src/.libs/librados.so ./.libs/librados.so ./.libs/libcls_rgw_client.a ./.libs/libcls_log_client.a ./.libs/libcls_statelog_client.a ./.libs/libcls_timeindex_client.a ./.libs/libcls_user_client.a ./.libs/libcls_replica_log_client.a ./.libs/libcls_lock_client.a ./.libs/libcls_refcount_client.a ./.libs/libcls_version_client.a -lcurl -lexpat -lfcgi /usr/local/lib/libldap.so ./.libs/libglobal.a ./.libs/libcommon.a -lboost_thread -latomic_ops -lboost_random -luuid -lpthread -lcryptopp -lm -lexecinfo -lboost_iostreams -lboost_system -Wl,-rpath -Wl,/usr/local/lib
./.libs/librgw.so: undefined reference to `rgw::RGWFileHandle::FLAG_NONE'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
Makefile:16057: recipe for target 'ceph_rgw_multiparser' failed